clients/imodelhub/src/IModelHubBackend.ts

```diff
@@ -302,22 +301,26 @@ export class IModelHubBackend implements BackendHubAccess {
     return val;
   }
 
-  public async downloadV1Checkpoint(arg: CheckpointArg): Promise<ChangesetIndexAndId> {
-    const checkpoint = arg.checkpoint;
+  public async downloadV1Checkpoint(arg: DownloadCheckpointArg): Promise<ChangesetIndexAndId> {
     let checkpointQuery = new CheckpointQuery().selectDownloadUrl();
-    checkpointQuery = checkpointQuery.precedingCheckpoint(checkpoint.changeset.id);
-    const accessToken = await this.getAccessToken(checkpoint);
-    const checkpoints = await this.iModelClient.checkpoints.get(accessToken, checkpoint.iModelId, checkpointQuery);
+    checkpointQuery = checkpointQuery.precedingCheckpoint(arg.changeset.id);
+    const accessToken = await this.getAccessToken(arg);
+    const checkpoints = await this.iModelClient.checkpoints.get(accessToken, arg.iModelId, checkpointQuery);
     if (checkpoints.length !== 1)
       throw new IModelError(BriefcaseStatus.VersionNotFound, "no checkpoints not found");
 
-    const cancelRequest: any = {};
-    const progressCallback: ProgressCallback = (progress: ProgressInfo) => {
-      if (arg.onProgress && arg.onProgress(progress.loaded, progress.total!) !== 0)
-        cancelRequest.cancel?.();
+    const progressCallback: ProgressCallback = (progress) => {
+      if (arg.progressCallback && progress.total)
+        arg.progressCallback(progress.loaded, progress.total);
     };
 
+    const cancelRequest: any = {};
+    const removeCancelListener = arg.cancelSignal?.addListener(() => cancelRequest.cancel?.());
+
     await this.iModelClient.checkpoints.download(accessToken, checkpoints[0], arg.localFile, progressCallback, cancelRequest);
+    if (removeCancelListener)
```

Comment:
Just FYI, you could also call it w/ optional chaining operator: `removeCancelListener?.();`

EDIT: nvm, already used at L318, stay consistent :)